Add issue templates

peteruithoven opened the pull request:
First example of https://github.com/elementary/os/issues/209

Goal is to improve the quality of issues, for example making sure people include all the relevant information from the get go. 

Inspired by:
- Github's default issue templates
- @kgrubb proposal
- [Signal's issue template](https://github.com/signalapp/Signal-Desktop/issues/new)

Tried to include as many existing resources as possible. 

I'm hoping to copy these templates to almost all repo's with minor customizations.

danrabbit commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,72 @@
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: bug
```
I don't think we want to use this label since everything in the issue tracker that a user reports can be described as a bug

cassidyjames replied:
Would it be evil to use the Incomplete label until a maintainer/somebody comes in and properly confirms/labels it?

peteruithoven replied:
We'll except for feature requests... 
But yeah bug is used way more often so I get why that might just add noise.

danrabbit commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,72 @@
+-->
+
+## Prerequisites
+- [ ] I have searched open and closed issues for duplicates.
```
Do we need this anymore since GitHub includes search when creating a title now?

cassidyjames replied:
I don't think we need it. I know some third-party clients might not search for issues ahead of time (i.e. an Android app or whatever) but that doesn't really seem like what we'd be catering to with issue templates anyway.

peteruithoven replied:
I would mainly include it to make people think twice. It kind of reiterate that people should really do this. 

Also these relates issues list only looks at the title and as we all know there are usually many ways to word things. When searching for existing issues I usually try 2 or 3 terms. 

In some cases it's useful to search through all the issues of the elementary org. For example when people open an issue in the nightlight indicator repo which is a duplicate of something in the nightlight settings repo. But I'm not sure how to explain that and if that applies often enough.

danrabbit commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,43 @@
+name: Feature request
+about: Suggest an idea for this project
+title: ''
+labels: enhancement
```
We use this label for feature requests.

```suggestion
labels: Priority: Wishlist
```

I wonder if we should also add the "needs design" label here by default?

cassidyjames replied:
@danrabbit yeah it makes sense that a feature request/wishlist item would require design input.

jlnr commented:
What do you think about linking to [Screen Recorder](https://appcenter.elementary.io/com.github.mohelm97.screenrecorder/) right from the template? It's a bit unfair because there several screen recording and GIF conversion apps on AppCenter, but the alternative is that users have to app hunting just to report a bug.

peteruithoven commented:
I've improved the default labels. Added `Needs design` to Feature requests and removed `bug` from bug report.

Even though I love Screen recorder it seems weird to recommend a specific non core application.

cassidyjames commented:
@peteruithoven it sounds like we can add these to the new `.github` repo, actually, and have them be the default for all repos. We should move them there, and then yeah I'm fine starting here since it'll be easier to update.

danrabbit commented:
Moved to https://github.com/elementary/.github/pull/1

peteruithoven commented:
Thanks for picking this up! 
It's awesome that github supports these org wide settings.
